opentelemetry-user-events-trace/src/trace/reentrant_spanprocessor.rs

```diff
@@ -0,0 +1,54 @@
+use std::fmt::Debug;
+use std::sync::{Arc, Mutex};
+
+use opentelemetry::Context;
+use opentelemetry_sdk::trace::Span;
+use opentelemetry_sdk::trace::SpanExporter;
+use opentelemetry_sdk::{error::OTelSdkResult, trace::SpanData};
+
+/// This export processor exports without synchronization.
+/// This is currently only used in user_events span exporter, where we know
+/// that the underlying exporter is safe under concurrent calls.
+
+#[derive(Debug)]
+pub struct ReentrantSpanProcessor<T: SpanExporter> {
+    exporter: Arc<Mutex<T>>,
```

Comment:
lets add a TODO to remove this one upstream publishes new version as it'll no longer need mutable self for export.

Reply:
Done.